### PR TITLE
feat(cloudwatch): add Bedrock Agent alarm coverage check

### LINKED ISSUE #12630: [New Check]: CloudWatch alarms cover Bedrock Agent metrics
### Existing check search

- [x] I have searched existing issues, Prowler Hub, and the public roadmap, and this check does not already exist.

### Provider

AWS

### New provider name

_No response_

### Service or product area

cloudwatch

### Suggested check name

cloudwatch_bedrock_agent_alarms_configured

### Context and goal

- Source control: `FS-11 — Agent Rate Alarms`.
- Semantic requirement: Verifies CloudWatch alarms exist for agent invocation rates.
- Valid cloud-posture assertion: No check evaluates AWS/Bedrock/Agents invocation or throttling alarms. The proposed check evaluates a durable AWS control-plane property with deterministic resource-level or regional posture semantics; it does not treat optional product adoption or manual governance evidence as failure.
- Resource scope: Each AWS region with Bedrock Agents, evaluated as one regional posture result.
- Initiative tracker: https://github.com/prowler-cloud/prowler/issues/12597

### Expected behavior

- Resource or scope to evaluate: Each AWS region with Bedrock Agents, evaluated as one regional posture result.
- PASS when: at least one enabled alarm directly or through metric math references AWS/Bedrock/Agents invocation or throttling metrics.
- FAIL when: agents exist but no qualifying alarm exists.
- MANUAL when: Required detail or policy data is unavailable or indeterminate after collection; never emit a false PASS for unreadable in-scope data.
- No resources: Return no findings. For regional/account-level posture checks, return no finding when the audited service has no relevant resources in that region.
- Error behavior: Preserve Prowler's normal service-error reporting and do not convert AccessDenied, unsupported-region responses, or partial inventory into PASS.

### References

- Immutable AWS source snapshot: https://github.com/aws-samples/sample-aiml-security-assessment/tree/0e2c5b48a6016a77e9d376bc17f41ef95cb2635d
- Source implementation for `FS-11`: https://github.com/aws-samples/sample-aiml-security-assessment/tree/0e2c5b48a6016a77e9d376bc17f41ef95cb2635d
- Audited Prowler snapshot: https://github.com/prowler-cloud/prowler/tree/2b81fdcc04c298bc476086b4082bd8fa4a9f1dd5
- Required AWS APIs/collector inputs: `cloudwatch:DescribeAlarms; existing Bedrock Agent inventory`.
- Full initiative and classification rationale: https://github.com/prowler-cloud/prowler/issues/12597

### Suggested severity

Medium

### Additional implementation notes

- Collector change: Extend CloudWatch alarm metric parsing for direct and metric-math namespace references.
- Dependencies: None beyond the stated service collector change.
- Emit one finding per resource unless the scope above explicitly defines a regional/account-level posture result.
- Add metadata and implementation files following the AWS check conventions, plus collector/model updates where required.
- Acceptance tests must cover PASS, FAIL, unreadable/indeterminate MANUAL behavior, empty inventory, API error handling, and all stated edge conditions.
- Tests must prove the exact AWS response fields above and ensure unrelated resources/configurations are not reported.
- Keep this CheckID atomic; do not combine it with adjacent assertions that have different resources, APIs, or remediation.

### Validation evidence required

Before this implementation can be considered ready for merge:

- The pull request must complete a CodeRabbit review, and all actionable findings must be addressed or explicitly resolved with technical justification.
- The contributor must provide execution evidence for one **PASS** case and one **FAIL** case using representative AWS resources. The evidence must include the executed Prowler command and the resulting finding output, demonstrating the expected status and `status_extended` message.
- All execution evidence must be censored before publication. Account IDs, resource names, ARNs, credentials, tokens, endpoints, and any other sensitive or identifying information must be visibly masked or removed from screenshots, logs, and command output.
- Unit tests are required, but they do not replace this runtime evidence.

### PR DESCRIPTION
**Pull request by haxonittech**
### Context

Fixes #12630.

Bedrock Agent alarms were easy to miss because the CloudWatch collector only kept its filtered alarm view and did not retain which metric queries an alarm actually watched. That could hide a valid alarm or, with metric math, count an unused metric source.

I also reviewed #12683. It adds a different Bedrock runtime alarm check and touches the same collector. This PR keeps the existing namespace fields compatible and adds the structured references and regional error state needed for the Agent check.

### Description

- Keep a complete CloudWatch metric-alarm inventory alongside the existing filtered view.
- Resolve direct metrics and only the metric-math queries reachable from `ReturnData=true` outputs.
- Match enabled `AWS/Bedrock/Agents` alarms for `InvocationCount` or `InvocationThrottles`, including the documented Agent dimensions.
- Return one PASS, FAIL, or MANUAL result per region with Bedrock Agents, and no result when there are no agents.
- Keep partial inventory and API errors separate so unreadable data cannot turn into PASS.
- Add the service dependency, metadata, docs, changelog fragment, and regression coverage.

No provider permission update is needed; this uses the existing Bedrock Agent inventory and CloudWatch `DescribeAlarms` collector permissions.

### Steps to review

1. Check the collector tests for direct alarms, nested metric math, unused metric sources, account IDs, resource filtering, and regional errors.
2. Check the new check tests for PASS, FAIL, MANUAL, empty inventory, regional isolation, disabled alarms, unrelated metrics, dimensions, and partial scans.
3. Run:

```bash
uv run pytest -q tests/providers/aws/services/cloudwatch/cloudwatch_service_test.py tests/providers/aws/services/cloudwatch/cloudwatch_bedrock_agent_alarms_configured/cloudwatch_bedrock_agent_alarms_configured_test.py
```

Post-rebase result: `58 passed`.

The full CloudWatch and Bedrock service suites also passed locally (`416 passed`), along with `make lint-sdk`, `uv lock --check`, Black, flake8, pylint, Bandit, vulture, YAML validation, actionlint, and zizmor.

### Runtime evidence

The redacted real-AWS PASS and FAIL runs required by #12630 are still pending an authorized sandbox account. I did not use or inspect any local AWS account from this environment, so this PR is staying in draft until that evidence is added.

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [x] This feature is listed in the open issues.
- [x] The issue is assigned to me.
- [x] I reviewed the open pull requests and confirmed there is no PR implementing the same Agent-alarm outcome.

</details>

- [x] The code is covered by tests.
- [x] Comments and docstrings follow the repository style.
- [x] Backport was reviewed and is not needed for this new check.
- [x] README impact was reviewed; no README change is needed.
- [x] A Prowler changelog fragment is included.

#### SDK/CLI

- New check: Yes
- Provider permission update: No

#### UI / API / MCP Server

Not applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

## Summary by CodeRabbit

- **New Features**
  - Added a check to verify CloudWatch alarms monitor Amazon Bedrock Agent invocation and throttling activity.
  - Findings now evaluate alarm coverage by AWS Region and identify missing or insufficient monitoring.
  - Improved detection of alarms using metric math, cross-account metrics, and detailed metric dimensions.
  - Scanning reports manual review when regional inventory or service access is incomplete.

- **Documentation**
  - Updated unit-testing guidance to include the Bedrock–CloudWatch service relationship.

### CODE REVIEW

**Comment by haxonittech** [comment]
@coderabbitai review pls, ty

File: prowler/providers/aws/services/cloudwatch/cloudwatch_service.py [hunk 1]
```diff
@@ -357,11 +455,20 @@ def _list_tags_for_resource(self, log_group):
             )
 
 
+class MetricReference(BaseModel):
```
[the remark this answers is not in the document]
Comment (haxonittech) [reply prowler/providers/aws/services/cloudwatch/cloudwatch_service.py:458]: added all three in 562ba571e; 58 focused tests + lint are still green. should be sorted now

**Comment by haxonittech** [comment]
@coderabbitai review pls, round 2

**Comment by haxonittech** [comment]
@coderabbitai review
